src/porepy/numerics/ad/operators.py

```diff
@@ -488,24 +486,20 @@ def _parse_readable(self, op: "Operator") -> str:
             operator_str = "*"
         elif tree.op == Operation.div:
             operator_str = "/"
-        elif tree.op in [Operation.evaluate, Operation.apply]:
-            # TODO: This has not really been tested.
+        elif tree.op == Operation.evaluate:
```

Comment:
I'm not sure I agree we should delete the last else, covering unknown operations (it should raise an error rather than print a statement, though). Two reasons:
1. Someone may try to hack the Ad framework, and somehow make tree with a not recognized error message. Due to the complexity of this code, I prefer to raise explicit errors, rather than letting the hacker suffer the consquences.
2.  More likely than 1., we will implement new operations, but forget to add them to this type of parsing. If so, a reminder in the form of an error would be good.

Reply:
Ah, you rather let operator_str be None, and give a special message below? If so, please document shortly.